contracts/escrow/escrow.sol

```diff
@@ -0,0 +1,279 @@
+	/// @param _newMultisig The address which is added as the new multisig.
+	/// @dev Can only be initiated by the current multisig.
+	event NewMultisig(address indexed _initiator, address indexed _newMultisig);
+
+	/// @notice Emitted when the release timestamp is updated.
+	/// @param _initiator The address which initiated this event to be emitted.
+	/// @param _releaseTimestamp The updated release timestamp for the withdraw.
+	event TokenReleaseUpdated(address indexed _initiator, uint256 _releaseTimestamp);
+
+	/// @notice Emitted when the deposit limit is updated.
+	/// @param _initiator The address which initiated this event to be emitted.
+	/// @param _depositLimit The updated deposit limit.
+	event TokenDepositLimitUpdated(address indexed _initiator, uint256 _depositLimit);
+
+	/// @notice Emitted when a new token deposit is done by User.
+	/// @param _initiator The address which initiated this event to be emitted.
+	/// @param _amount The amount of token deposited.
+	event TokenDeposit(address indexed _initiator, uint256 _amount);
+
+	/// @notice Emitted when we reach the token deposit limit.
+	event DepositLimitReached();
+
+	/// @notice Emitted when a token withdraw is done by Multisig.
+	/// @param _initiator The address which initiated this event to be emitted.
+	/// @param _amount The amount of token withdrawed.
+	event TokenWithdrawByMultisig(address indexed _initiator, uint256 _amount);
+
+	/// @notice Emitted when a new token deposit is done by Multisig.
+	/// @param _initiator The address which initiated this event to be emitted.
+	/// @param _amount The amount of token deposited.
+	event TokenDepositByMultisig(address indexed _initiator, uint256 _amount);
+
+	/// @notice Emitted when a token withdraw is done by User.
+	/// @param _initiator The address which initiated this event to be emitted.
+	/// @param _amount The amount of token withdrawed.
+	event TokenWithdraw(address indexed _initiator, uint256 _amount);
+
+	/* Modifiers */
+
+	modifier onlyMultisig() {
+		require(msg.sender == multisig, "Only Multisig can call this.");
+		_;
+	}
+
+	modifier checkStatus(Status s) {
+		require(status == s, "The contract is not in the right state.");
+		_;
+	}
+
+	modifier checkRelease() {
+		require(releaseTime != 0 && releaseTime <= block.timestamp, "The release time has not started yet.");
+		_;
+	}
+
+	/* Functions */
+
+	/**
+	 * @notice Setup the required parameters.
+	 * @param _SOV The SOV token address.
+	 * @param _multisig The owner of the tokens & contract.
+	 * @param _releaseTime The token release time, zero if undecided.
+	 * @param _depositLimit The amount of tokens we will be accepting.
+	 */
+	constructor(
+		address _SOV,
+		address _multisig,
+		uint256 _releaseTime,
+		uint256 _depositLimit
+	) public {
+		require(_SOV != address(0), "Invalid SOV Address.");
+		require(_multisig != address(0), "Locked token & contract owner address invalid.");
+
+		SOV = IERC20(_SOV);
+		multisig = _multisig;
```

Comment:
Multisig Event added in commit 499f2e4

SOV does not need an event, and the rest as it could be dummy values, for now, should not have an event in the constructor.